cmd/capibmadm/cmd/powervs/key/create.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package key
+
+import (
+	"context"
+
+	"github.com/spf13/cobra"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/log"
+
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/iam"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/powervs"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/options"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/utils"
+)
+
+type keyCreateParams struct {
+	key     string
+	keyName string
+}
+
+// CreateSSHKeyCommand - child command of 'key' to create an ssh key.
+func CreateSSHKeyCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create an ssh-key in the PowerVS environment.",
+		Example: `
+# Create an ssh key.
+export IBMCLOUD_API_KEY=<api-key>
+capibmadm powervs key create --name <key-name> --key "<ssh key>" --service-instance-id <service-instance-id> --zone <zone>`,
+	}
+
+	var keyCreateParam keyCreateParams
+	cmd.Flags().StringVar(&keyCreateParam.key, "key", "", "SSH RSA key string within a double quotation marks. For example, \"ssh-rsa AAA... \".")
+	cmd.Flags().StringVar(&keyCreateParam.keyName, "name", "", "The name of the key.")
+	_ = cmd.MarkFlagRequired("key")
+	_ = cmd.MarkFlagRequired("key-name")
+
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		if err := createSSHKey(cmd.Context(), keyCreateParam); err != nil {
+			return err
+		}
+		return nil
+	}
+	return cmd
+}
+
+func createSSHKey(ctx context.Context, keyCreateParam keyCreateParams) error {
+	logger := log.Log
+	logger.Info("Creating ssh key...")
+	auth := iam.GetIAMAuth()
+	accountID, _ := utils.GetAccountID(ctx, auth)
+	sess, err := powervs.NewPISession(accountID, options.GlobalOptions.PowerVSZone, options.GlobalOptions.Debug)
+	if err != nil {
+		return err
+	}
+	c := instance.NewIBMPIKeyClient(ctx, sess, options.GlobalOptions.ServiceInstanceID)
```

Comment:
instead of c lets use something meaningful name like keyClient.

Reply:
Addressed.